modules/systemd-journal/tests/test_systemd_journal.c

```diff
@@ -25,16 +25,30 @@
 #include "test-source.h"
 #include "journald-helper.c"
 #include "journal-reader.c"
-#include "testutils.h"
 #include "apphook.h"
-
-#define JOURNALD_TESTCASE(testfunc, ...) { testcase_begin("%s(%s)", #testfunc, #__VA_ARGS__); testfunc(__VA_ARGS__); testcase_end(); }
-
-#define TEST_PERSIST_FILE_NAME "test_systemd_journal.persist"
+#include <criterion/criterion.h>
 
 static gboolean task_called;
 static gboolean poll_triggered;
 
+static void
+_init_cfg_with_persist_file(const gchar *persist_file)
+{
+  main_thread_handle = get_thread_id();
+  configuration = cfg_new_snippet();
```

Comment:
I think it would be better to create a `GlobalConfig` object instead of using the global `configuration`.

Reply:
ok.

Reply:
Oh sorry, I did not see that this one has not been resolved yet. I'm marking this unresolved.

Reply:
If I use GlobalConfig object, one of the test case on my machine fails.
```
[FAIL] systemd_journal::test_journal_reader: CRASH!
```

Reply:
Yes, you have to pass your config object in that test:
```
test_source_new(configuration) // <-- here,
```

One way of doing that would be, if `_init_cfg_with_persist_file()` returned the created `GlobalConfig` object, so you can pass it to the constructor of the test source.

Reply:
Yes, Thanks @MrAnno

Reply:
@mehul-m-prajapati why does it crashes ? maybe there is some issue the test found this way.

Reply:
@Kokan : That might be the case. I am still debugging the issue.

Reply:
Maybe the global `configuration` is used somewhere in the test, that is not initialised causing null pointer dereference.

Reply:
@Kokan : You are correct. I can see below code in test-source.c
```
static gboolean
__init(LogPipe *s)
{
  TestSource *self = (TestSource *)s;
  self->reader = journal_reader_new(configuration, self->journald_mock);
  journal_reader_options_defaults(&self->options);
  if (self->current_test_case && self->current_test_case->init)
    {
      self->current_test_case->init(self->current_test_case, self, self->journald_mock, self->reader, &self->options);
    }
  journal_reader_options_init(&self->options, configuration, "test");
```

Now, how can i pass local GlobalConfig object in a __init callback of test-source.c ?

Reply:
The `LogPipe`'s init function signature only accepts `LogPipe`, so passing anything can be done via `self`. Luckily your use case have been foreseen by others and `log_pipe_get_config` might help you out :)

Reply:
@Kokan : Thanks for the suggestion. I have modified the patch. Please review the same.